scipy/stats/__init__.py

```diff
@@ -105,6 +108,7 @@
    semicircular      -- Semicircular
    skewnorm          -- Skew normal
    t                 -- Student's T
+   template_gen      -- Template (Generator)
```

Comment:
If we add this and the other one as public functionality, we should find some better names. `_gen` suffixes have been an implementation details and I guess they better stay like this. Also `template_gen` is too generic and non-descriptive.